File: 1044-find-common-characters/find-common-characters.java
class Solution {
    public List<String> commonChars(String[] words) {
        int[][] arr = new int[26][words.length];
        for(int i=0;i<words.length;i++)
        {
            for(int j=0;j<words[i].length();j++)
            {
                arr[words[i].charAt(j)-'a'][i]++;
            }
        }
        List<String> ans = new ArrayList<>();
        for(int i=0;i<26;i++)
        {
            int f=Integer.MAX_VALUE;
            for(int j=0;j<words.length;j++)
            {
                f=Math.min(f,arr[i][j]);
            }
            while(f-->0)
            {
                ans.add(Character.toString('a'+i));
            }
        }
        return ans;
    }
}

File: 1044-find-common-characters/find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        vector<vector<int>> m(26, vector<int> (words.size(),0));
        for(int i=0;i<words.size();i++)
        {
            string s=words[i];
            for(int j=0;j<s.size();j++)
            {
                int ch=s[j]-'a';
                m[ch][i]++;
            }
        }
        vector<string> ans;
        for(int i=0;i<26;i++)
        {
            int f=INT_MAX;
            for(int j=0;j<words.size();j++)
            {
                f=min(f,m[i][j]);
            }
            while(f--)
            {
                char ch='a'+i;
                string str="";
                str+=ch;
                ans.push_back(str);
            }
        }
        return ans;
    }
};